### PR TITLE
Run terraform linter in CI

### LINKED ISSUE #3: Run test (linter) in CI
The test that was added in #2  should run on every PR. We could do that via Github actions.

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

Fixes #3

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# validate any terraform file changes
+name: Terraform lint
+
+env:
+  TERRAFORM_VERSION: "1.2.5"
+  TERRAGRUNT_VERSION: "0.38.6"
+
+on:
+  push:
```
Comment (irbekrm) [inline .github/workflows/lint.yaml]: Suggestion: it should be sufficient to only run on 'pull_request' and not on 'push'. The workflow will then be triggered when a pull request is opened and also [when new update for the pull request is pushed](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request:~:text=occurs.%20For%20example%2C-,if%20no%20activity%20types%20are%20specified%2C%20the%20workflow%20runs%20when%20a%20pull%20request%20is%20opened%20or%20reopened%20or%20when%20the%20head%20branch%20of%20the%20pull%20request%20is%20updated,-.%20For%20activity%20related)

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# validate any terraform file changes
+name: Terraform lint
+
+env:
+  TERRAFORM_VERSION: "1.2.5"
+  TERRAGRUNT_VERSION: "0.38.6"
+
+on:
```
Comment (irbekrm) [inline .github/workflows/lint.yaml:8]: Suggestion: could we also add [`workflow_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)? I usually find it very useful when testing something or re-running on flakes.

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+jobs:
+  terraform-lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (irbekrm) [inline .github/workflows/lint.yaml]: Question: why do we need to set `fetch-depth` to 0? I would have thought that the default of 1 would be sufficient?

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thanks @lonelyCZ this looks good, I haven't tested it, but read through the workflow and added a couple nits. Happy to merge otherwise, we can then trigger it manually if you add the `workflow_dispatch` event I think.

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (lonelyCZ) [reply .github/workflows/lint.yaml:8]: Thanks for your suggestions, I have updated it.

**Review by irbekrm — APPROVED** [review]
Thank you @lonelyCZ this is great work!

/lgtm

**Comment by irbekrm** [comment]
I manually triggered the workflow and it ran and passed https://github.com/cert-manager/testing-addons/runs/7516270636?check_suite_focus=true 🥳

**Pull request merged**